layers.go

```diff
@@ -823,33 +823,45 @@ func (r *layerStore) Diff(from, to string, options *DiffOptions) (io.ReadCloser,
 	if options != nil && options.Compression != nil {
 		compression = *options.Compression
 	}
+	maybeCompressReadCloser := func(rc io.ReadCloser) (io.ReadCloser, error) {
+		if compression == archive.Uncompressed {
```

Comment:
This name is meaningless to me.  Could you add a comment explaining what this function is doing?